src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -1267,26 +1267,6 @@ void mf::XWaylandSurface::prep_surface_spec(ProofOfMutexLock const&, msh::Surfac
         }
     }
 
-    auto scale_size_clamped = [inv_scale](auto& optional_prop)
-        {
-            if (optional_prop)
-            {
-                using ValueType = typename std::remove_reference<decltype(optional_prop.value())>::type;
-                using UnderlyingType = typename ValueType::ValueType;
-                auto constexpr raw_max = std::numeric_limits<UnderlyingType>::max();
-
-                double const double_value = optional_prop.value().as_value() * inv_scale;
-                optional_prop = std::max(ValueType{1}, ValueType{std::min(double_value, double{raw_max})});
-            }
-        };
-
-    scale_size_clamped(mods.width);
-    scale_size_clamped(mods.height);
-    scale_size_clamped(mods.min_width);
-    scale_size_clamped(mods.min_height);
-    scale_size_clamped(mods.max_width);
-    scale_size_clamped(mods.max_height);
-
```

Comment:
We should not be dropping the clamping. (Rewriting using `std::clamp` could be a thing though)

Reply:
That makes sense now.